common/numerated/src/tree.rs

```diff
@@ -77,10 +77,10 @@ use scale_info::{
 /// let voids = tree.voids(..).map(RangeInclusive::from).collect::<Vec<_>>();
 /// assert_eq!(voids, vec![i32::MIN..=0, 4..=4, 8..=i32::MAX]);
 ///
-/// // AndNot iterator: iterate over intervals from `tree` which are not in `other_tree`.
-/// let other_tree: IntervalsTree<i32> = [3, 4, 5, 7, 8, 9].iter().collect();
-/// let and_not: Vec<_> = tree.and_not_iter(&other_tree).map(RangeInclusive::from).collect();
-/// assert_eq!(and_not, vec![1..=2, 6..=6]);
+/// // Difference iterator: iterate over intervals from `tree` which are not in `other_tree`.
+/// let other_tree: IntervalsTree<i32> = [3, 4, 5, 7, 8, 9].into_iter().collect();
+/// let difference: Vec<_> = tree.difference(&other_tree).map(RangeInclusive::from).collect();
+/// assert_eq!(difference, vec![1..=2, 6..=6]);
 /// ```
 ///
 /// # Possible panic cases
```

Comment:
https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L91
Add trait bound `T: Numerated`

Reply:
Same for interval

Reply:
It will to force to write `T: Numerated` for every impl of each corresponding type

Reply:
What the reason?